Source/RpmAvatarCreator/Public/NARF/Buttons/RpmButtonBase.h

```diff
@@ -0,0 +1,43 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Components/Button.h"
+#include "RpmButtonBase.generated.h"
+
+class UImage;
+/**
+ * 
+ */
+UCLASS(Abstract, Blueprintable)
+class RPMAVATARCREATOR_API URpmButtonBase : public UUserWidget
+{
+	GENERATED_BODY()
+
+public:
+	// Function to set the icon of the button
+	UFUNCTION(BlueprintCallable, Category = "Ready Player Me/Creator")
+	void SetIcon(UTexture2D* IconTexture);
+
+	// Function to set the color of the button
+	UFUNCTION(BlueprintCallable, Category = "Ready Player Me/Creator")
+	void SetColor(const FLinearColor& NewColor);
```

Comment:
This function should be removed, it is specific only to the color button. This goes against the interface segregation principle.